### ISSUE TITLE
It Seems Something's Wrong with the GPU Usage of the Matcher

### ISSUE THREAD

**Issue by 3942368** [3DOM-FBK/deep-image-matching#83]
I experimented with superpoint + lightglue and aliked + lightglue.

The GPU on my local machine is a 1060 6GB, and the cloud environment on lightning.ai is using an L4 GPU.

On my local machine, the matcher speed is around 1.56 it/s, but in the cloud, it only reaches 3.52 it/s.
I tested with around 600 images at a resolution of 1280x720.
The command I used for testing was:
-p aliked+lightglue --skip_reconstruction -s sequential --overlap 30 
It seems like the matchers aren’t utilizing the GPU properly.
Isn't this speed an issue?

this is local gpu speed(1060 6G)
![image](https://github.com/user-attachments/assets/2d7c391d-58f8-4290-8d32-7aa36661ea69)
this is cloud gpu(L4)
![image](https://github.com/user-attachments/assets/4883f418-fd23-4eb9-945c-a0b6495dd501)

![image](https://github.com/user-attachments/assets/9bf3de33-6a05-4abf-8ea8-f948450d9350)

**Comment by lcmrl** [comment]
Hi, thanks for reporting, could you check inside the conda environment prepared for DIM if cuda is really available?
```
> python
> import torch
> torch.cuda.is_available()
```

**Comment by 3942368** [comment]
![image](https://github.com/user-attachments/assets/a0512d7b-3be8-412a-a6c0-15665acaf77a)
here you are

**Comment by 3942368** [comment]
> Hi, thanks for reporting, could you check inside the conda environment prepared for DIM if cuda is really available?
> 
> ```
> > python
> > import torch
> > torch.cuda.is_available()
> ```

![image](https://github.com/user-attachments/assets/d8ad67ee-4b36-4316-9328-dcded59ae788)
![image](https://github.com/user-attachments/assets/caca4a71-7508-409e-9d90-7c7739d594ac)
This is the result from running on the Colab environment set up on your side. It was a T4 GPU with the aliked+lightglue configuration. Of course, PyTorch is properly applied (since it's your Colab code). However, the speed in matching is only 1.27s/it. It seems like the GPU is not being fully utilized.

**Comment by lcmrl** [comment]
Hi, thanks for reporting. Did you solve the issue? A solution to significantly decrease the matching time is to use `GeometricVerification.MAGSAC` instead of `GeometricVerification.PYDEGENSAC` that is much slower

**Comment by lcmrl** [comment]
Hope this solved the issue, thanks for the feedback, feel free to reopen or collaborate on the repo

**Comment by 3942368** [comment]
I'm sorry I saw this too late.

When I experimented locally, I confirmed that the speed improved by about two times under the GeometricVerification.PYDEGENSAC condition.

I wanted to also check the GeometricVerification.NONE (no geometric verification) condition to see the speed difference, but several errors occurred, so I am still investigating.

Your solution was effective. Thank you very much.
